source/UF/SequentialColimits.lagda

```diff
@@ -0,0 +1,532 @@
+  = ⌜ sequential-cocone-identity-characterization 𝓑 𝓑' ⌝⁻¹
+
+\end{code}
+
+Given a sequential cocone over X and a map X → Y there is a canonical assignment to
+a sequential cocone over Y.
+
+\begin{code}
+
+module _ (𝓐 : type-sequence 𝓤)
+         (X : 𝓥 ̇) (Y : 𝓣 ̇)
+       where
+
+ canonical-map-to-sequential-cocone : sequential-cocone 𝓐 X
+                                    → (X → Y)
+                                    → sequential-cocone 𝓐 Y
+ canonical-map-to-sequential-cocone (h , H) u =
+  ((λ n → u ∘ h n) , λ n → ∼-ap-∘ u (H n))
+
+\end{code}
+
+A sequential cocone over X is universal if the above map is an equivalence for any Y.
+Such a sequential cocone is said to be the sequential colimit of a type sequence.
+
+\begin{code}
+
+ Sequential-Colimit-Universal-Property : (𝓧 : sequential-cocone 𝓐 X)
+                                       → 𝓤 ⊔ 𝓥 ⊔ 𝓣 ̇
+ Sequential-Colimit-Universal-Property 𝓧 =
+  is-equiv (canonical-map-to-sequential-cocone 𝓧)
+
+\end{code}
+
+We now give a construction of the sequential colimit in terms of the pushout.
+This construction follows 26.2 in Introduction to Homotopy Type Theory (link above).
+
+The sequential colimit A∞ can be constructed as the pushout of the following diagram
+
+                     [id , σ]
+          Σ A + Σ A ------------> Σ A
+              |                    |
+   [id , id]  |                    | inrr
+              |                    |
+             Σ A ----------------> A∞
+                       inll
+
+where σ (n , x) = (n + 1 , a n x).
+
+\begin{code}
+
+module _ (𝓐@(A , a) : type-sequence 𝓤)
+         (X : 𝓣 ̇)
+       where
+
+ σ : Σ A → Σ A
+ σ (n , x) = (succ n , a n x)
+
+ f : Σ A + Σ A → Σ A
+ f = cases id id
+
+ g : Σ A + Σ A → Σ A
+ g = cases id σ
+
+ private
+  index : Σ A → ℕ
+  index = pr₁
+
+  element-at : ((n , x) : Σ A) → A n
+  element-at = pr₂
+
+ module _ (push-ex : pushouts-exist f g)
+           where
+
+  open pushouts-exist push-ex
+
+  sequential-colimit : 𝓤 ̇
+  sequential-colimit = pushout
+
+\end{code}
+
+We provide the sequential cocone structure for the sequential colimit. Notice since the
+pushout is formed on a coproduct we get two equations from the canonical pushout gluing.
+These equations will prove useful throughout the development.
+
+\begin{code}
+
+  ι : (n : ℕ) → A n → sequential-colimit
+  ι n x = inrr (n , x)
+
+  glue-inl : ((n , x) : Σ A) → inll (n , x) ＝ ι n x
+  glue-inl (n , x) = glue (inl (n , x))
+
+  glue-inr : ((n , x) : Σ A) → inll (n , x) ＝ ι (succ n) (a n x)
+  glue-inr (n , x) = glue (inr (n , x))
+
+  K : (n : ℕ) → ι n ∼ ι (succ n) ∘ a n
+  K n x = glue-inl (n , x) ⁻¹ ∙ glue-inr (n , x)
+
+  sequential-colimit-is-cocone : sequential-cocone 𝓐 sequential-colimit
+  sequential-colimit-is-cocone = (ι , K)
+
+\end{code}
+
+We will quickly provide names and a technical lemma that will prove useful later when
+applying a map to some of the above paths.
+
+\begin{code}
+
+  ap-glue-inl : (u : sequential-colimit → X)
```

Comment:
Again, I don't see why you need these nearly identical synonyms. Maybe I didn't make my earlier suggestion clear. What I meant was to abbreviate some larger terms like `ap u (glue-inl (n , x))` with something short like `p` in the longer proofs below.